api.py

```diff
@@ -0,0 +1,277 @@
+        crawl.addBoth(self.on_success)
+        return jsonify({
+            "data": {
+              "status": "running",
+              "spider": spider.name,
+              "_id": spider_id
+            }
+        }), 200
+
+    def on_success(self, data):
+        self.database._close_all_spiders()
+
+    def close_spider(self, spider_id):
+        for crawl in self.process.crawlers:
+            if crawl.spider.uuid == uuid:
+                crawl.stop()
+                return jsonify({
+                    "data":
+                        {"status": "success", "_id": spider_id}
+                    }), 200
+        return '', 400
+
+    def list_crawls(self):
+        crawls = self.process.crawlers
+        running_spiders = []
+        for crawl in crawls:
+            start_time = crawl.stats.get_value('start_time')
+            spider = {
+                '_id':
+                    crawl.spider.uuid,
+                'spider':
+                    crawl.spider.name,
+                'strart_time':
+                    start_time,
+                'total_time':
+                    str(datetime.now() - start_time),
+                'item_dropped':
+                    crawl.stats.get_value('item_dropped_count'),
+                'item_scraped':
+                    crawl.stats.get_value('item_scraped_count'),
+                'total_requests':
+                    crawl.stats.get_value('downloader/request_count'),
+            }
+
+            running_spiders.append(spider)
+        finished_spiders = []
+        for spider in self.database.get_finished_crawls():
+            finished_spiders.append(spider)
+        spiders = {"crawling": running_spiders, "finished": finished_spiders}
+        return jsonify({"data": {"spiders": spiders}}), 200
+
+    def stop(self):
+        self.process.stop()
+        return jsonify({"data": {"status": "success"}}), 200
+
+    def export_db(self):
+        articles_rows = self.database.get_articles()
+        articles = []
+        now = datetime.now()
+        for row in articles_rows:
+            articles.append({
+                'title': row[0],
+                'file_hash': row[1],
+                'url': row[2],
+            })
+        json_file = tempfile.NamedTemporaryFile()
+        json_file.write(json.dumps(articles).encode('utf-8'))
+        json_file.seek(0)
+        return send_file(
+            json_file,
+            mimetype='application/json',
+            as_attachment=True,
+            attachment_filename=f'export-{now}.json'
+        )
+
+    def import_db(self):
+        if request.files:
+            data_file = request.files.get('file', None)
+            if data_file.filename == '':
+                return 'Filename must not be blank', 400
+            if data_file.content_type == 'application/json':
+                json_file = data_file.stream.read()
+            else:
+                return 'File format is not json.', 400
```

Comment:
probably 415, https://stackoverflow.com/questions/11973813/http-status-code-for-unaccepted-content-type-in-request